### PR TITLE
Simple index pages for docc

### PR DESCRIPTION
**Pull request by PeterAdams-A**
Motivation:

An index page ties all the other documentation together

Modifications:

Add index pages for the library targets.
Correct a few minor errors in the main docs.

Result:

A more joined up documentation experience.

### CODE REVIEW

File: Sources/NIOExtras/Docs.docc/index.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# NIOExtras
```
Comment (Lukasa) [inline Sources/NIOExtras/Docs.docc/index.md:1]:
```suggestion
# ``NIOExtras``
```

File: Sources/NIOHTTPCompression/Documentation.docc/index.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+# NIOHTTPCompression
```
Comment (Lukasa) [inline Sources/NIOHTTPCompression/Documentation.docc/index.md:1]:
```suggestion
# ``NIOHTTPCompression``
```

File: Sources/NIOSOCKS/Docs.docc/index.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# NIOSOCKS
```
Comment (Lukasa) [inline Sources/NIOSOCKS/Docs.docc/index.md:1]:
```suggestion
# ``NIOSOCKS``
```

**Review by Lukasa — CHANGES_REQUESTED** [review]
A few notes in the diff, and I recommend changing `Documentation.docc` to `Docs.docc` in `NIOHTTPCompression` for consistency.

**Review by Lukasa — APPROVED** [review]

**Review by MurdoqZodoq — COMMENTED** [review]
Dr. Miyka’El Zodoq

**Pull request merged**